src/orion/algo/axoptimizer.py

```diff
@@ -0,0 +1,344 @@
+            space,
+            seed=seed,
+            n_initial_trials=n_initial_trials,
+            extra_objectives=extra_objectives,
+            constraints=constraints,
+        )
+
+    def seed_rng(self, seed):
+        """Seed the state of the random number generator.
+
+        :param seed: Integer seed for the random number generator.
+
+        .. note:: Ax does not promise deterministic trials generation and only
+           similar trials generation.
+        """
+        if self._client_state is not None:
+            raise RuntimeError(
+                f"{self.__class__.__name__} does not support reseeding an instantiated client"
+            )
+        self.seed = seed
+
+    @property
+    def state_dict(self):
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state_dict = copy.deepcopy(super().state_dict)
+        # AxClient.to_json_snapshot() seams to be more like the current internal
+        # state of the client than independent a snapshot
+        state_dict["_client_state"] = copy.deepcopy(self._client_state)
+        state_dict["_trials_map"] = copy.deepcopy(self._trials_map)
+
+        return state_dict
+
+    def set_state(self, state_dict):
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        super().set_state(copy.deepcopy(state_dict))
+        self._client_state = copy.deepcopy(state_dict.get("_client_state"))
+        self._trials_map = copy.deepcopy(state_dict.get("_trials_map"))
+
+    @contextlib.contextmanager
+    def get_client(self):
+        """Instantiate a new `AxClient` from previous snapshot"""
+        if self._client_state is not None:
+            # Copy client state because `from_json_snapshot` modifies it...
+            client = AxClient.from_json_snapshot(copy.deepcopy(self._client_state))
+        else:
+            client = AxClient(
+                random_seed=self.seed,
+                enforce_sequential_optimization=False,
+                verbose_logging=False,
+            )
+
+            client.create_experiment(
+                parameters=orion_space_to_axoptimizer_space(self.space),
+                choose_generation_strategy_kwargs={
+                    "num_initialization_trials": self.n_initial_trials,
+                    "max_parallelism_override": self.max_trials,
+                },
+                objectives={
+                    "objective": ObjectiveProperties(minimize=True),
+                    **{
+                        o: ObjectiveProperties(minimize=True)
+                        for o in self.extra_objectives
+                    },
+                },
+                outcome_constraints=self.constraints,
+            )
+
+        yield client
+
+        self._client_state = client.to_json_snapshot()
+
+    def suggest(self, num):
+        """Suggest a `num`ber of new sets of parameters.
```

Comment:
```suggestion
        """Suggest a number of new sets of parameters.
```
This has caused an error while running `tox -e docs` for algo BOHB. Maybe you are having the same issue here. :crossed_fingers: